src/Compilers/Core/Portable/SymbolDisplay/AbstractSymbolDisplayVisitor_Minimal.cs

```diff
@@ -47,11 +48,21 @@ protected bool NameBoundSuccessfullyToSameSymbol(INamedTypeSymbol symbol)
             var type1 = GetSymbolType(normalSymbol);
             var type2 = GetSymbolType(typeOnlySymbol);
 
-            return
+            return (
                 type1 != null &&
                 type2 != null &&
                 type1.Equals(type2) &&
-                typeOnlySymbol.Equals(symbol.OriginalDefinition);
+                typeOnlySymbol.Equals(symbol.OriginalDefinition)) || IsGenericTypeArgument(type1, symbol);
```

Comment:
It's not clear why we compare `typeOnlySymbol` with the type of the symbol from `LookupSymbols` or against it's type arguments. That wouldn't handle cases where the type returned from `LookupNamespacesAndTypes` is not part of the type of the symbol from `LookupSymbols`. For instance:
```
namespace ConsoleApplication1
{
    class Program
    {
        static C<A> B { get; set; }
        static C<B> A { get; set; }

        static void Main(string[] args)
        {
            B = new $$
        }
    }
    class A { }
    class B { }
    class C<T> { }
}
```
Would we get the expected result with `return typeOnlySymbol.Equals(symbol.OriginalDefinition);`?